### PR TITLE
[Merge on CLI release] Templating documentation

### PR DESCRIPTION
**Pull request by CameronMcWilliam**
## Checklist

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Description
Added documentation for the new templating capability of the `stack package` command.

## Related Issue

Related to: [appsody/appsody#528](https://github.com/appsody/appsody/pull/528)

#issue-number

### CODE REVIEW

File: content/docs/stacks/modify.md [hunk 1]
```diff
@@ -16,6 +16,58 @@ The `image` directory contains files for building the stack image. This image co
...
-The `templates` directory contains one or more starter applications that are given to the user when they initialise their projects. 
+The `templates` directory contains one or more starter applications that are given to the user when they initialize their projects.
 
 Whilst modifying templates, consider if the functionality may be better placed in the stack image itself to affect all templates.
+
+## Templating
+
+Often in a stack, there are common values that are used across the image and template.  It can be laborious to manually go through a stack and make changes to these values in every place they occur, especially if those values change frequently, such as the version number.  Templating offers a solution to this problem.
+
+Templating is applied when the `stack package` command is called.  To use templating in your stack, wrap the variables you would like to use with `{{ }}`.
```
Comment (neeraj-laad) [inline content/docs/stacks/modify.md]: I think we should say we support standard Go Templating to be clear and provide a link to their documentation.

File: content/docs/stacks/modify.md [hunk 2]
```diff
@@ -16,6 +16,58 @@ The `image` directory contains files for building the stack image. This image co
...
+
+**Example usage:**
+
+`This is {{.stack.name}}, running version: {{.stack.version}}.`
+
+**Note:** Do not use templating for a `README`. As templating only takes place during the `stack package` command, the `README` on the GitHub page for your stack would have the template variables as opposed to the values associated with them.
```
Comment (neeraj-laad) [inline content/docs/stacks/modify.md]: very minor - I've seen a recommendation to use a specific pencil icon to highlight notes.

File: content/docs/stacks/modify.md [hunk 3]
```diff
@@ -16,6 +16,58 @@ The `image` directory contains files for building the stack image. This image co
...
 ## Modifying stack templates
-The `templates` directory contains one or more starter applications that are given to the user when they initialise their projects. 
+The `templates` directory contains one or more starter applications that are given to the user when they initialize their projects.
 
 Whilst modifying templates, consider if the functionality may be better placed in the stack image itself to affect all templates.
+
+## Templating
+
+Often in a stack, there are common values that are used across the image and template.  It can be laborious to manually go through a stack and make changes to these values in every place they occur, especially if those values change frequently, such as the version number.  Templating offers a solution to this problem.
+
```
Comment (neeraj-laad) [inline content/docs/stacks/modify.md]: Perhaps add a line to describe our solution to the problems described above.  Something on the lines of..

Stack creators can declare values in a central place (stack.yaml) and use templating constructs to refer to them in rest of the files for the stack. This way values can be changed in one place and as always in sync.

File: content/docs/stacks/modify.md (hunk 1 above)
Comment (neeraj-laad) [inline content/docs/stacks/modify.md]: How about `Templates are translated into their values before a stack is packaged?`

File: content/docs/stacks/modify.md [hunk 4]
```diff
@@ -16,6 +16,58 @@ The `image` directory contains files for building the stack image. This image co
...
+Templating is applied when the `stack package` command is called.  To use templating in your stack, wrap the variables you would like to use with `{{ }}`.
+
+**Example usage:**
+
+`This is {{.stack.name}}, running version: {{.stack.version}}.`
+
+**Note:** Do not use templating for a `README`. As templating only takes place during the `stack package` command, the `README` on the GitHub page for your stack would have the template variables as opposed to the values associated with them.
+
+### Built-in templating
```
Comment (neeraj-laad) [inline content/docs/stacks/modify.md]: Built-in templating variables

File: content/docs/stacks/modify.md [hunk 5]
```diff
@@ -16,6 +16,58 @@ The `image` directory contains files for building the stack image. This image co
...
+| `.stack.tag`              | The `tag` value from Docker image.                                                    |
+| `.stack.maintainers`      | The `maintainers` list from `stack.yaml`.                                             |
+| `.stack.version`          | The `version` value from `stack.yaml`.                                                |
+| `.stack.semver.major`     | The `version` major value from `stack.yaml`.                                          |
+| `.stack.semver.minor`     | The `version` minor value from `stack.yaml`.                                          |
+| `.stack.semver.patch`     | The `version` patch value from `stack.yaml`.                                          |
+| `.stack.semver.majorminor`| The `version` major and minor values from `stack.yaml`.                               |
+| `.stack.image.namespace`  | The `image-namespace` from user defined image-namespace flag, default is `dev.local`. |
+
+### Custom templating
```
Comment (neeraj-laad) [inline content/docs/stacks/modify.md]: Custom templating variables

File: content/docs/stacks/modify.md [hunk 6]
```diff
@@ -16,6 +16,58 @@ The `image` directory contains files for building the stack image. This image co
...
+
+If you would like to use your own custom variables you can declare a `templating-data` map in your `stack.yaml`:
```
Comment (neeraj-laad) [inline content/docs/stacks/modify.md]: The map should only contain key: value pairs?

File: content/docs/stacks/modify.md (hunk 1 above)
Comment (CameronMcWilliam) [reply content/docs/stacks/modify.md]: I like that wording, do we still want to mention that it takes place during the `stack package` command, or is that implied with that wording?

File: content/docs/stacks/modify.md (hunk 2 above)
Comment (CameronMcWilliam) [reply content/docs/stacks/modify.md]: I was using **Note:** as a placeholder as I believe @ddunn2 is adding styling for `>` for notes. How do I add this pencil icon?

**Pull request merged**